### PR TITLE
Feature spring cloud alibaba sentinel zuul

### LINKED ISSUE #58: [Feature] Add spring-cloud-alibaba-sentinel-zuul module
Add Sentinel spring-cloud-zuul adapter which provides `Service` and `API Path` level rate limiter.

### PR DESCRIPTION
**Pull request by tigerMoon**
### Describe what this PR does / why we need it

add module spring cloud alibaba sentinel zuul
### Does this pull request fix one issue?
Fixes #58 
### Describe how you did it

make this module as spring boot starer module.
### Describe how to verify it

run in local spring boot project

### CODE REVIEW

File: spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+@ConditionalOnProperty(prefix = PREFIX, name = "enabled", havingValue = "true")
+public class SentinelZuulAutoConfiguration {
+
+    @Autowired
+    private Environment environment;
+
+    public static final String PREFIX = "spring.cloud.alibaba.sentinel.zuul";
+
+    @Bean
+    public SentinelZuulProperties sentinelZuulProperties() {
```
Comment (xiaolongzuo) [inline spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java]: Why don't use @ConfigurationProperties and @EnableConfigurationProperties?

**Comment by gaoyaqiu** [comment]
什么时候能再spring cloud alibaba中使用sentinel-zuul 呢

File: spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java (hunk 1 above)
Comment (fangjian0423) [reply spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java]: SentinelZuulProperties is not annotated by @ConfigurationProperties

File: spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/listener/FallBackProviderListener.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+/**
+ * @author tiger
+ */
+public class FallBackProviderListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(FallBackProviderListener.class);
+
+    @EventListener(classes = ApplicationStartedEvent.class)
```
Comment (fangjian0423) [inline spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/listener/FallBackProviderListener.java]: Maybe using SmartInitializingSingleton is better, ApplicationStartedEvent is not compatibility in Greenwich.

**Comment by fangjian0423** [comment]
> 什么时候能再spring cloud alibaba中使用sentinel-zuul 呢

Next version will be supported. It is preparing now.

File: spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java (hunk 1 above)
Comment (tigerMoon) [reply spring-cloud-alibaba-sentinel-zuul/src/main/java/org/springframework/cloud/alibaba/sentinel/zuul/SentinelZuulAutoConfiguration.java]: we use sentinel-zuul-adapter as a dependency which can be used in independent. and it`s not a spring project.

**Pull request merged**